lib/rubocop/target_finder.rb

```diff
@@ -100,6 +97,31 @@ def find_files(base_dir, flags)
       Dir.glob(pattern, flags).select { |path| FileTest.file?(path) }
     end
 
+    private
+
+    def ls_git_files(base_dir)
+      return if `sh -c 'command -v git'`.empty?
+
+      output, _error, status = Open3.capture3(
+        'git', 'ls-files', '-z', base_dir,
+        '--exclude-standard', '--others', '--cached', '--modified'
+      )
```

Comment:
I’m wondering if this logic shouldn’t be extracted to a class, to encapsulate the dependency on `git` and `sh`? That would also make it possible to test this functionality in isolation.

Reply:
I don't see a need for this (and advantages). Isolation — yes, but why? I think, integration (like real life) tests are better. And they're not complicated.

Reply:
I imagine a regression test for #6622 would be easier to write for an isolated class. But yes, integration tests (for happy path) are also necessary.